### PR TITLE
components/form: add `FormPersonalDetails` component

### PR DESCRIPTION
**Pull request by maceksimon**
Add `FormPersonalDetails` component for the first step of the challenge registration.

* Add component
* Add basic tests for form
* Add translation strings

### CODE REVIEW

File: src/components/form/FormPersonalDetails.vue [hunk 1]
```diff
@@ -0,0 +1,233 @@
...
+ *
+ * @see [Figma Design](https://www.figma.com/file/L8dVREySVXxh3X12TcFDdR/Do-pr%C3%A1ce-na-kole?type=design&node-id=4858%3A102976&mode=dev)
+ */
+
+import { defineComponent, nextTick } from 'vue';
+
+// components
+import FormFieldTextRequired from 'components/global/FormFieldTextRequired.vue';
+
+export type FormPersonalDetailsFields = {
```
Comment (tmszi) [inline src/components/form/FormPersonalDetails.vue:32]: Shouldn't the types be in the types directory?

Comment (maceksimon) [reply src/components/form/FormPersonalDetails.vue]: Yes, they should. Corrected.

File: src/components/form/FormPersonalDetails.vue [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+    formValues: {
+      type: Object as () => FormPersonalDetailsFields,
+      required: true,
+    },
+  },
+  emits: ['update:formValues'],
+  setup(props, { emit }) {
+    const personalDetails: FormPersonalDetailsFields = props.formValues;
+
+    const newsletterOptions = [
```
Comment (tmszi) [inline src/components/form/FormPersonalDetails.vue:50]: Missing type?

File: src/components/form/FormPersonalDetails.vue [hunk 3]
```diff
@@ -0,0 +1,226 @@
...
+        value: 'events',
+      },
+      {
+        label: 'form.personalDetails.labelNewsletterMobility',
+        value: 'mobility',
+      },
+    ];
+
+    const genderOptions = [
+      {
```
Comment (tmszi) [inline src/components/form/FormPersonalDetails.vue:70]: Missing type?

**Comment by tmszi** [comment]
If required form fields is not filled in, Continue button should not allow to switch to another stepper page, right?

File: src/components/form/FormPersonalDetails.vue (hunk 2 above)
Comment (maceksimon) [reply src/components/form/FormPersonalDetails.vue:50]: Added.

File: src/components/form/FormPersonalDetails.vue (hunk 3 above)
Comment (maceksimon) [reply src/components/form/FormPersonalDetails.vue:70]: Added.

**Comment by maceksimon** [comment]
> If required form fields is not filled in, Continue button should not allow to switch to another stepper page, right?

Yes, certainly. I did aim to make this PR small so I left that out just now, but I guess it also makes sense to include it to make the feature complete.

**Pull request merged**